test/dag_test.go

```diff
@@ -62,6 +62,12 @@ func TestDAGPipelineRun(t *testing.T) {
 		t.Fatalf("Failed to create echo Task: %s", err)
 	}
 
+	// Make sure the Pipeline has been created (wait for it)
+	if err := WaitForTaskCreated(c, "echo-task", "TaskCreated"); err != nil {
+		t.Errorf("Error waiting for Task echo-task to be created: %s", err)
+		t.Fatal("Pipeline execution failed, Task echo-task has not been created")
```

Comment:
nit: Don't need a `t.Error` and a `t.Fatal`, you could just combine these into one descriptive `t.Fatal`.